src/WeETH.sol

```diff
@@ -76,14 +82,64 @@ contract WeETH is ERC20Upgradeable, UUPSUpgradeable, OwnableUpgradeable, ERC20Pe
         return eETHAmount;
     }
 
+    /// @notice Requires the spender to be whitelisted before calling {ERC20PermitUpgradeable-permit}
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public override {
+        require(whitelistedSpender[spender], "weETH: spender not whitelisted"); 
+    
+        super.permit(owner, spender, value, deadline, v, r, s);
+    }
+
     //--------------------------------------------------------------------------------------
     //-------------------------------  INTERNAL FUNCTIONS  ---------------------------------
     //--------------------------------------------------------------------------------------
 
+    /// @dev Authorizes the upgrade of the contract to a new implementation by the owner
+    /// @param newImplementation The address of the new contract implementation
     function _authorizeUpgrade(
         address newImplementation
     ) internal override onlyOwner {}
 
+
+    /// @notice Require the recipient to not be blacklisted before calling {ERC20Upgradeable-_transfer}
+    function _transfer(
+        address from,
+        address to,
+        uint256 amount
+    ) internal override {
+        require(blacklistedRecipient[to] == false, "weETH: recipient blacklisted");
```

Comment:
the same here

Reply:
Initial Gas Cost: 29848
Gas Cost with check only on `to`: 32100
Gas Cost if we add checks on both `to` and `from`: 34321

Reply:
thx